.github/workflows/test.yaml

```diff
@@ -57,6 +58,12 @@ jobs:
         uses: actions/checkout@v3
       - name: Unit test
         run: make test
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: go
+      - name: Perform CodeQL Analysis
```

Comment:
we need a autobuild step here. example: https://github.com/kubernetes/release/blob/master/.github/workflows/codeql-analysis.yml#L65

Reply:
Sorry Sertac, was travelling.. Was travelling. Will review all below - thank you

Reply:
Seems to work on my machine (12 mins for complete run - one vulnerability warning